cmd/remove.go

```diff
@@ -45,18 +44,19 @@ func init() {
 			// the resource to delete is set as the source
 			raw.src = args[0]
 
-			// infer the location of the delete
 			srcLocationType := inferArgumentLocation(raw.src)
-			if srcLocationType == common.ELocation.Blob() {
-				raw.fromTo = common.EFromTo.BlobTrash().String()
-			} else if srcLocationType == common.ELocation.File() {
-				raw.fromTo = common.EFromTo.FileTrash().String()
-			} else if srcLocationType == common.ELocation.BlobFS() {
-				raw.fromTo = common.EFromTo.BlobFSTrash().String()
-			} else {
-				return fmt.Errorf("invalid source type %s to delete. azcopy support removing blobs/files/adls gen2", srcLocationType.String())
+			if raw.fromTo == "" {
+				// infer the location of the delete
+				if srcLocationType == common.ELocation.Blob() {
+					raw.fromTo = common.EFromTo.BlobTrash().String()
+				} else if srcLocationType == common.ELocation.File() {
+					raw.fromTo = common.EFromTo.FileTrash().String()
+				} else if srcLocationType == common.ELocation.BlobFS() {
+					raw.fromTo = common.EFromTo.BlobFSTrash().String()
+				} else {
+					return fmt.Errorf("invalid source type %s to delete. azcopy support removing blobs/files/adls gen2", srcLocationType.String())
+				}
```

Comment:
If `raw.fromTo != ""`,  i.e. from-to is provided by the user, it makes sense to verify that the destination is a trash. Otherwise we throw appropriate error saying `fromto.To()` isn't trash. Let's have that check in frontend layer for that as well.